src/engine/strat_engine/names.rs

```diff
@@ -236,3 +236,40 @@ pub fn validate_name(name: &str) -> StratisResult<()> {
     }
     Ok(())
 }
+
+#[cfg(test)]
+mod tests {
+
+    use engine::strat_engine::names::validate_name;
+
+    #[test]
+    pub fn test_validate_name() {
+        assert!(validate_name(&'\u{0}'.to_string()).is_err());
+        assert!(validate_name("./some").is_err());
+        assert!(validate_name("../../root").is_err());
+        assert!(validate_name("/").is_err());
+        assert!(validate_name("\u{1c}\u{7}").is_err());
+        assert!(validate_name("./foo/bar.txt").is_err());
+        assert!(validate_name(".").is_err());
+        assert!(validate_name("..").is_err());
+        assert!(validate_name("/dev/sdb").is_err());
+        assert!(validate_name("").is_err());
+        assert!(validate_name("/").is_err());
+        assert!(validate_name(" leading_space").is_err());
+        assert!(validate_name("trailing_space ").is_err());
+        assert!(validate_name("\u{0}leading_null").is_err());
+        assert!(validate_name("trailing_null\u{0}").is_err());
+        assert!(validate_name("middle\u{0}_null").is_err());
+        assert!(validate_name("\u{0}multiple\u{0}_null\u{0}").is_err());
+
+        assert!(validate_name(&'\u{10fff8}'.to_string()).is_ok());
+        assert!(validate_name("*< ? >").is_ok());
+        assert!(validate_name("...").is_ok());
+        assert!(validate_name("ok.name").is_ok());
+        assert!(validate_name("ok name with spaces").is_ok());
+        assert!(validate_name("\\\\").is_ok());
+        assert!(validate_name("\u{211D}").is_ok());
+        assert!(validate_name("☺").is_ok());
+        assert!(validate_name("ok_name").is_ok());
```

Comment:
``` rust
assert!(validate_name("𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏\
                       𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏").is_err());
assert!(validate_name("𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏\
                       𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏𐌏").is_ok());
```

Reply:
```
iter::repeat("𐌏").take(<whatever the correct number is>).collect::<Vec<_>>().join("")
```
would beat having to count the number of 𐌏s.

Reply:
Or slightly better

```
iter::repeat("𐌏").take(<whatever the correct number is>).collect::<Vec<_>>().concat()
```

Reply:
Each character '𐌏' is encoded to 4 bytes, thus 64 characters = 256 which is too long, 63 works.

``` rust
assert!(validate_name(iter::repeat("𐌏").take(64).collect::<Vec<_>>().concat()).is_err());
assert!(validate_name(iter::repeat("𐌏").take(63).collect::<Vec<_>>().concat()).is_ok());
```

Reply:
Ok - I added these tests.